lib/rubyserial/posix.rb

```diff
@@ -23,7 +22,7 @@ def initialize(address, baude_rate=9600, data_bits=8, parity=:none)
 
     @config = build_config(baude_rate, data_bits, parity)
 
-    err = RubySerial::Posix.tcsetattr(@fd, RubySerial::Posix::TCSANOW, @config)
+    err = RubySerial::Posix.tcsetattr(@fd, RubySerial::Posix::TCSAFLUSH, @config)
```

Comment:
This changed from `TCSANOW` to `TCSAFLUSH`, without the `tcflush` call below, seems to do the trick.